src/map_field.cpp

```diff
@@ -1046,10 +1046,17 @@ void field_processor_fd_fire( const tripoint &p, field_entry &cur, field_proc_da
             smoke += static_cast<int>( windpower / 5 );
             if( cur.get_field_intensity() > 1 &&
                 one_in( 200 - cur.get_field_intensity() * 50 ) ) {
+                here.spawn_item( p, "ash", 1, rng( 10, 1000 ) );
                 if( p.z > 0 ) {
-                    // We're in the air
+                    // We're in the air. Need to invalidate the furniture otherwise it'll cause problems
+                    here.furn_set( p, f_null );
```

Comment:
Is there a way to bash the furniture and put the result of the bash on the z-level below? That might be a nicer solution.

Reply:
Bash the furniture, yes. There's *too many* functions to do that.

Figuring out if the results are placeable below probably requires a new function. I'll take a look at it sometime soon